common/src/main/java/org/opensearch/ml/common/transport/prompt/MLUpdatePromptInput.java

```diff
@@ -0,0 +1,210 @@
+     *
+     * @param input Abstract class that describes Stream Input
+     * @throws IOException thrown if an I/O exception occurred while reading the object from StreamInput
+     */
+    public MLUpdatePromptInput(StreamInput input) throws IOException {
+        this.name = input.readOptionalString();
+        this.description = input.readOptionalString();
+        this.version = input.readOptionalString();
+        this.prompt = input.readMap(s -> s.readString(), s -> s.readString());
+        this.tags = input.readList(StreamInput::readString);
+        this.tenantId = input.readOptionalString();
+        this.lastUpdateTime = input.readOptionalInstant();
+    }
+
+    /**
+     * Write MLUpdatePromptInput object to StreamOutput
+     *
+     * @param output Abstract class that describes Stream Output
+     * @throws IOException thrown if an I/O exception occurred while writing the object to StreamOutput
+     */
+    @Override
+    public void writeTo(StreamOutput output) throws IOException {
+        output.writeOptionalString(name);
+        output.writeOptionalString(description);
+        output.writeOptionalString(version);
+        output.writeMap(prompt, StreamOutput::writeString, StreamOutput::writeString);
+        output.writeCollection(tags, StreamOutput::writeString);
+        output.writeOptionalString(tenantId);
+        output.writeOptionalInstant(lastUpdateTime);
+    }
+
+    /**
+     * Write MLUpdatePromptInput object to XContent
+     *
+     * @param builder XContentBuilder
+     * @param params Parameters
+     * @return XContentBuilder
+     * @throws IOException thrown if an I/O exception occurred while writing the object to XContent
+     */
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (name != null) {
+            builder.field(PROMPT_NAME_FIELD, name);
+        }
+        if (description != null) {
+            builder.field(PROMPT_DESCRIPTION_FIELD, description);
+        }
+        if (version != null) {
+            builder.field(PROMPT_VERSION_FIELD, version);
+        }
+        if (prompt != null) {
+            builder.field(PROMPT_PROMPT_FIELD, prompt);
+        }
+        if (tags != null) {
+            builder.field(PROMPT_TAGS_FIELD, tags);
+        }
+        if (tenantId != null) {
+            builder.field(TENANT_ID_FIELD, tenantId);
+        }
+        if (lastUpdateTime != null) {
+            builder.field(LAST_UPDATED_TIME_FIELD, lastUpdateTime.toEpochMilli());
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    /**
+     * Parse XContent field values and Create MLUpdatePromptInput object
+     *
+     * @param parser XContentParser
+     * @return MLUpdatePromptInput
+     * @throws IOException if an I/O exception occurred while parsing the XContent
+     */
+    public static MLUpdatePromptInput parse(XContentParser parser) throws IOException {
```

Comment:
I think it worth a validation/allowlist to restrict which fields are allowed to be updated, and what are the allowed value format for each field. For example, the tags cannot exceed 128 letters, etc.